tests/Unit/TestHelpers.hpp

```diff
@@ -289,39 +289,45 @@ std::string get_output(const Container& c) {
 
 /*!
  * \ingroup TestingFramework
- * \brief Alternative to CAPTURE that prints more digits.
+ * \brief Alternative to Catch's CAPTURE that prints more digits.
  */
 #define CAPTURE_PRECISE(variable)                                    \
   INFO(std::scientific << std::setprecision(18) << #variable << ": " \
                        << (variable));
 
 /*!
  * \ingroup TestingFramework
- * \brief Calculates the derivative of an Invocable at a point x, represented by
- * an array of doubles in the domain of map with a sixth-order finite difference
+ * \brief Calculates the derivative of an Invocable at a point x - represented
+ * by
+ * an array of doubles - in the domain of map with a sixth-order finite
+ * difference
  * method.
  *
  * \details Intended for use with CoordinateMaps taking the domain {xi,eta,zeta}
  * to the range {x,y,z}. This function calculates the derivative along the
- * direction given by the only non-zero element of dx.
+ * direction given by `direction` with a step size of `h`.
  *
- * \requires an array dx with a single non-zero coordinate: this
- * will be the direction along the domain in which the derivative is taken.
+ * \requires direction be between 0 and VolumeDim
  */
 template <typename Invocable, size_t VolumeDim>
 std::array<double, VolumeDim> numerical_derivative(
     const Invocable& map, const std::array<double, VolumeDim>& x,
-    const std::array<double, VolumeDim>& dx) {
-  ASSERT(VolumeDim - 1 ==
-             static_cast<size_t>(std::count(dx.begin(), dx.end(), 0.0)),
-         "Can only take derivative along a single logical axis at a time.");
+    const size_t direction, const double delta) {
+  ASSERT(0 <= direction and direction < VolumeDim,
+         "Trying to take derivative along axis " << direction);
+
+  const auto dx = [direction, delta]() {
+    std::array<double, VolumeDim> d;
```

Comment:
This is technically undefined behavior you're relying on. You can use `make_array<VolumeDim>(0.0);`